orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/LogoutService.java

```diff
@@ -100,47 +101,57 @@ private APIGatewayProxyResponseEvent generateLogoutResponse(
                 302, "", Map.of(ResponseHeaders.LOCATION, logoutUri.toString()), null);
     }
 
-    private void destroySessions(Session session) {
-        for (String clientSessionId : session.getClientSessions()) {
-            clientSessionService
-                    .getClientSession(clientSessionId)
-                    .flatMap(
-                            t ->
-                                    t.getAuthRequestParams().get("client_id").stream()
-                                            .findFirst()
-                                            .flatMap(dynamoClientService::getClient))
-                    .ifPresent(
-                            clientRegistry ->
-                                    backChannelLogoutService.sendLogoutMessage(
-                                            clientRegistry,
-                                            session.getEmailAddress(),
-                                            configurationService.getInternalSectorURI()));
-            LOG.info("Deleting Client Session");
-            clientSessionService.deleteStoredClientSession(clientSessionId);
-        }
-        LOG.info("Deleting Session");
-        sessionService.deleteStoredSession(session.getSessionId());
+    private void destroySessions(
+            Optional<Session> session,
+            Optional<OrchSessionItem> orchSession,
+            Optional<String> clientId,
+            Optional<AccountIntervention> accountIntervention) {
+        session.ifPresent(
```

Comment:
I'm not a big fan of this. Only in `handleLogout` are the session and orchSession optional, not in the other 2, and it means here, we end up with a lot of nesting.

Reply:
I agree it's ugly, but personally I think this is fine as a stopgap until we delete the session logic. We could add a separate destroyOrchSession method, but it might be easy to delete the deleteSession method without moving over the client session logic.